build.gradle

```diff
@@ -18,46 +24,85 @@ repositories {
     jcenter()
 }
 
+publishing {
+    publications {
+        customLibrary(MavenPublication) {
+            from components.java
+        }
+    }
+
+    repositories {
+        maven {
+            name = 'mt.edu.um.cf2.jgribx'
+            url = layout.buildDirectory.dir("repo")
+        }
+    }
+}
+
 sourceSets {
-	main {
-		resources.srcDirs = ["src/main/java","src/main/resources"]
-	}
-	test {
-		resources.srcDirs = ["src/test/java", "src/test/resources"]
-	}
+    main {
+        resources.srcDirs = ["src/main/java", "src/main/resources"]
+    }
+    test {
+        resources.srcDirs = ["src/test/java", "src/test/resources"]
+    }
 }
 
 configurations {
     extraLibs
 }
 
 test {
-		testLogging {
-    		events 'started', 'passed'
-		}
+    testLogging {
+        events 'started', 'passed'
+    }
+}
+
+configurations {
+    runtime
+    ajc
+    aspects
+    expandedJars.transitive = false
+    api {
+        extendsFrom(aspects)
+        extendsFrom(expandedJars)
+    }
+    testArtifacts.extendsFrom testRuntime
+    offlinePluginDependency.extendsFrom(compileOnly)
 }
 
 jar {
+    duplicatesStrategy = DuplicatesStrategy.EXCLUDE
     from {
-        configurations.extraLibs.collect { it.isDirectory() ? it : zipTree(it) }
+        // Expand all the project jars into the main jar so Raptor can find the CDI annotations
+        (configurations.expandedJars).collect { zipTree(it) }
```

Comment:
I think we can revert this back to what the original author had here; we never actually use the `expandedJars` configuration.